app/controllers/calendar_controller.rb

```diff
@@ -1,11 +1,51 @@
 class CalendarController < ApplicationController
+  helper_method :get_event, :event_ongoing?, :event_span
 
   def index
-    @days = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri']
+    week_start = Date.new(2016, 6, 11)
+    week_end = Date.new(2016, 6, 15)
+    @days = (week_start..week_end).to_a
+
     start_time = Time.now.at_beginning_of_day
     end_time = Time.now.at_end_of_day
     step = 30.minutes
-    @times = (start_time.to_i..end_time.to_i).step(step).collect { |time| Time.at time }
+    @times = time_interval(start_time, end_time, step)
+
+    @events = {}
+    @days.each do |day|
+      @events[day.wday] = []
+    end
+    Event.all.map do |event|
```

Comment:
I saw that guys prepare some cool scope `for_week` or sth like this. How about we would retrieve Events only for one week.